### DISCUSSION TITLE
Firebase app check uses com.google.android.gms.safetynet.safetyNet.

### DISCUSSION THREAD

**Discussion by tootisabz** [firebase/flutterfire#13025] [Show and tell]
Firebase app check uses safetyNet class which not support by google play. Google play store creates issue when uploaded android build to play store review. Please update it.

[✓] Flutter (Channel stable, 3.22.2, on macOS 14.5 23F79 darwin-arm64, locale en-AF)
    • Flutter version 3.22.2 on channel stable at /Users/macbookair/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 761747bfc5 (4 weeks ago), 2024-06-05 22:15:13 +0200
    • Engine revision edd8546116
    • Dart version 3.4.3
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/macbookair/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0-17.0.10b1087.21-11609105)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/xcode.app/Contents/Developer
    • Build 15E204a
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0-17.0.10b1087.21-11609105)

[✓] VS Code (version 1.90.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.90.0
    
    pubspec.yaml file:

  firebase_core: ^3.1.1
  firebase_app_check: ^0.3.0+2
  firebase_auth: ^5.1.1
  
  configurations.all {
    exclude group: 'com.google.android.gms', module: 'play-services-safetynet'
}
tried to remove it but it gives error during build release build

ERROR: R8: Missing class com.google.android.gms.safetynet.SafetyNet (referenced from: void com.google.firebase.appcheck.safetynet.internal.SafetyNetAppCheckProvider.lambda$initSafetyNetClient$0(com.google.android.gms.common.GoogleApiAvailability, android.content.Context, com.google.android.gms.tasks.TaskCompletionSource))
Missing class com.google.android.gms.safetynet.SafetyNetApi$AttestationResponse (referenced from: com.google.android.gms.tasks.Task com.google.firebase.appcheck.safetynet.internal.SafetyNetAppCheckProvider.exchangeSafetyNetAttestationResponseForToken(com.google.android.gms.safetynet.SafetyNetApi$AttestationResponse) and 1 other context)
Missing class com.google.android.gms.safetynet.SafetyNetClient (referenced from: void com.google.firebase.appcheck.safetynet.internal.SafetyNetAppCheckProvider.<init>(com.google.firebase.FirebaseApp, com.google.android.gms.safetynet.SafetyNetClient, com.google.firebase.appcheck.internal.NetworkClient, java.util.concurrent.Executor, java.util.concurrent.Executor, com.google.firebase.appcheck.internal.RetryManager) and 3 other contexts)

**Comment by kitterive** [comment]
I had same issue, even I updated firebase_auth to 5.3.1  and firebase_core to 3.6.0

**Comment by riccardocescon** [comment]
@tootisabz, @kitterive  did you manage to make this work?
Having the same issue even without using firebase_core
